### PR TITLE
TopologySpreadConstraint support

### PR DESCRIPTION
**Pull request by jfrconley**
Small update to add support for pod ToplogySpreadConstraints, as well as an associated feature gate.  The constraints are simply passed through to the generated stateful set similar to the affinity rules.

**Checklist**

* [x] I have added these changes to the changelog (or it's not applicable).

### CODE REVIEW

**Comment by jfrconley** [comment]
@davidwding Seems like I can't request reviews directly

File: apis/v1alpha1/cluster_types.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ type CrdbClusterSpec struct {
 	// (Optional) If specified, the pod's scheduling constraints
 	// +optional
 	Affinity *corev1.Affinity `json:"affinity,omitempty"`
+	// (Optional) If specified, the pod's topology spread constraints
+	// +optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints"`
```
Comment (davidwding) [inline apis/v1alpha1/cluster_types.go:111]: could you add `omitempty` to the json tag like affinity has?

File: pkg/resource/statefulset.go [hunk 2]
```diff
@@ -226,6 +226,10 @@ func (b StatefulSetBuilder) makePodTemplate() corev1.PodTemplateSpec {
 		pod.Spec.Tolerations = b.Spec().Tolerations
 	}
 
+	if utilfeature.DefaultMutableFeatureGate.Enabled(features.TopologySpreadRules) {
```
Comment (davidwding) [inline pkg/resource/statefulset.go:229]: could you also add this to pkg/resource/job.go? https://github.com/cockroachdb/cockroach-operator/blob/master/pkg/resource/job.go#L105

**Review by davidwding — COMMENTED** [review]
Hey @jfrconley thanks for the contribution! A few minor comments but looks like a good addition overall.

File: apis/v1alpha1/cluster_types.go (hunk 1 above)
Comment (jfrconley) [reply apis/v1alpha1/cluster_types.go]: Done

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 * Fixed operator crash loop when cockroachDBVersion is used.
 * Fix add custom annotations to the pod created by the job
 * Fix issue when sidecar container is injected to job pod
+* Add support for pod ToplogySpreadConstraint and associated feature gate
```
[the remark this answers is not in the document]
Comment (jfrconley) [reply CHANGELOG.md]: Done

File: pkg/resource/statefulset.go (hunk 2 above)
Comment (jfrconley) [reply pkg/resource/statefulset.go:229]: Done

**Review by davidwding — APPROVED** [review]

**Pull request merged**